empiric-package/empiric/core/client.py

```diff
@@ -1,28 +1,41 @@
-from empiric.core.const import NETWORK, ORACLE_CONTROLLER_ADDRESS
+from typing import Any, Optional, Tuple
+
+from empiric.core.config import get_config
+from empiric.core.errors import InvalidNetworkError
+from empiric.core.types import ADDRESS, TESTNET, Network
 from empiric.core.utils import str_to_felt
 from starknet_py.contract import Contract
 from starknet_py.net.gateway_client import GatewayClient
 
 
 class EmpiricClient:
-    def __init__(self, network=None, oracle_controller_address=None):
-        if network is None:
-            network = NETWORK
-        if oracle_controller_address is None:
-            oracle_controller_address = ORACLE_CONTROLLER_ADDRESS
+    oracle_controller_address: ADDRESS
+    oracle_controller_contract: Optional[ADDRESS]
 
+    def __init__(
+        self,
+        network: Network = TESTNET,
+        oracle_controller_address: Optional[ADDRESS] = None,
+    ):
         self.network = network
-        self.oracle_controller_address = oracle_controller_address
+        try:
+            self.config = get_config(network)()
+        except TypeError:
+            raise InvalidNetworkError(f"Invalid Network name: {network}")
+
+        self.oracle_controller_address = (
+            oracle_controller_address or self.config.ORACLE_CONTROLLER_ADDRESS
+        )
         self.oracle_controller_contract = None
 
     async def fetch_oracle_controller_contract(self):
         if self.oracle_controller_contract is None:
             self.oracle_controller_contract = await Contract.from_address(
                 self.oracle_controller_address,
-                GatewayClient(self.network),
+                GatewayClient(self.network, self.config.CHAIN_ID),
             )
 
-    async def get_decimals(self, key):
+    async def get_decimals(self, key) -> Any:
```

Comment:
I'd like to fill in these Any times with the correct values, wanted to get input on what type these should be

Reply:
This will be an int